ClangPowerTools/ClangPowerTools/ClangPowerToolsPackage.cs

```diff
@@ -266,7 +274,22 @@ public int OnAfterCloseSolution(object aPUnkReserved)
 
 
     #region Private Methods
-
+    private bool DoesSolutionContainCppProject()
+    {
+      DTE2 dte2 = Package.GetGlobalService(typeof(DTE)) as DTE2;
```

Comment:
Please use the `VsServiceProvider` class to get the services you need.

Reply:
Ok, done.